service/frontend/adminHandler.go

```diff
@@ -1479,6 +1483,188 @@ func (adh *AdminHandler) GetTaskQueueTasks(
+		tag.WorkflowID(execution.WorkflowId),
+		tag.WorkflowRunID(execution.RunId),
+	)
+
+	if execution.RunId == "" {
+		resp, err := adh.persistenceExecutionManager.GetCurrentExecution(ctx, &persistence.GetCurrentExecutionRequest{
+			ShardID:     shardID,
+			NamespaceID: namespaceID.String(),
+			WorkflowID:  execution.WorkflowId,
+		})
+		if err != nil {
+			return nil, adh.error(err, scope)
+		}
+		execution.RunId = resp.RunID
+	}
+
+	var warnings []string
+	var branchTokens [][]byte
+	var startTime, closeTime *time.Time
+	cassVisBackend := strings.Contains(adh.visibilityMgr.GetName(), cassandra.CassandraPersistenceName)
+
+	resp, err := adh.persistenceExecutionManager.GetWorkflowExecution(ctx, &persistence.GetWorkflowExecutionRequest{
+		ShardID:     shardID,
+		NamespaceID: namespaceID.String(),
+		WorkflowID:  execution.WorkflowId,
+		RunID:       execution.RunId,
+	})
+	if err != nil {
+		if common.IsContextCanceledErr(err) || common.IsContextDeadlineExceededErr(err) {
+			return nil, adh.error(err, scope)
+		}
+		// continue to deletion
+		warnMsg := "Unable to load mutable state when deleting workflow execution, " +
+			"will skip deleting workflow history and cassandra visibility record"
+		logger.Warn(warnMsg, tag.Error(err))
+		warnings = append(warnings, fmt.Sprintf("%s. Error: %v", warnMsg, err.Error()))
+	} else {
+		// load necessary information from mutable state
+		executionInfo := resp.State.GetExecutionInfo()
+		histories := executionInfo.GetVersionHistories().GetHistories()
+		branchTokens = make([][]byte, 0, len(histories))
+		for _, historyItem := range histories {
+			branchTokens = append(branchTokens, historyItem.GetBranchToken())
+		}
+
+		if cassVisBackend {
+			if resp.State.ExecutionState.State != enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
+				startTime = executionInfo.GetStartTime()
+			} else {
+				completionEvent, err := adh.getWorkflowCompletionEvent(ctx, shardID, resp.State)
+				if err != nil {
+					warnMsg := "Unable to load workflow completion event, will skip deleting visibility record"
+					adh.logger.Warn(warnMsg, tag.Error(err))
+					warnings = append(warnings, fmt.Sprintf("%s. Error: %v", warnMsg, err.Error()))
+				} else {
+					closeTime = completionEvent.GetEventTime()
+				}
+			}
+		}
+	}
+
+	if !cassVisBackend || (startTime != nil || closeTime != nil) {
+		// if using cass visibility, then either start or close time should be non-nil
+		// NOTE: the deletion is best effort, for sql and cassandra visibility implementation,
+		// we can't guarantee there's no update or record close request for this workflow since
+		// visibility queue processing is async. Operation can call this api again to delete again
+		// if this happens.
+		// For ES implementation, the TaskID will be used as version so no request will be applied
+		// after the deletion.
+		if err := adh.visibilityMgr.DeleteWorkflowExecution(ctx, &manager.VisibilityDeleteWorkflowExecutionRequest{
```

Comment:
This is not working right now since frontend ES client doesn't allow write operations.